irohad/main/impl/on_demand_ordering_init.cpp

```diff
@@ -171,38 +172,62 @@ namespace iroha {
             proposal_factory,
         std::shared_ptr<ametsuchi::TxPresenceCache> tx_cache,
         consensus::Round initial_round) {
+      auto delay = [reject_counter = static_cast<uint64_t>(0),
```

Comment:
What is the point to use a static cast? I've expected to use literals instead: `0u`.

Reply:
Nice trick with storing variables in closure but it seems that there is a need to add some comments about this fact.